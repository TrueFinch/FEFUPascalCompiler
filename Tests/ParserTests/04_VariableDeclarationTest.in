program a;
var
    e: function() : integer;
    ee: procedure();
    eee: function(i: integer) : integer;
    eeee: procedure(i: integer);
    eeeee: function(i, b: integer) : integer;
    eeeeee: procedure(i, b: integer);
    eeeeeee: function(i, b: integer; t: float) : integer;
    eeeeeeee: procedure(i, b: integer; t: float);
begin
end.