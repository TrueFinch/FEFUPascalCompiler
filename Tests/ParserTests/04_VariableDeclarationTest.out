                                                                                                                                            a                                                                                                                                            
                                                                                                                                            ┼                                                                                                                                            
                                                                                                                                        MainBlock                                                                                                                                        
                                                                                                                                  ┌─────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐         
                                                                                                                                 var                                                                                                                                   CompoundStatement 
       ┌──────────────┬───────────────────────┬─────────────────────────────┬──────────────────────────────┬──────────────────────┴────────┬──────────────────────────────────────────┬─────────────────────────────────────────────────────┐                                            
 SimpleVarDecl  SimpleVarDecl           SimpleVarDecl                 SimpleVarDecl                  SimpleVarDecl                   SimpleVarDecl                              SimpleVarDecl                                         SimpleVarDecl                                      
 ┌─────┼        ┌─────┴─┐       ┌─────────────┴──┐                ┌─────────┴──┐             ┌─────────────┴───┐                 ┌─────────┴───┐              ┌───────────────────────┴────┐                            ┌───────────────────┴────┐                                       
 e  function    ee  procedure  eee           function            eeee      procedure       eeeee           function            eeeeee      procedure       eeeeeee                     function                      eeeeeeee                procedure                                   
       ┼                                    ┌────┴─────────┐                   ┼                          ┌────┴─────────┐                     ┼                            ┌──────────────┴────┬──────────────┐                       ┌─────────┴─────────┐                             
    integer                         FormalParamSection  integer        FormalParamSection         FormalParamSection  integer          FormalParamSection           FormalParamSection  FormalParamSection  integer            FormalParamSection  FormalParamSection                    
                                    ┌──────┬┴                          ┌──────┬┴                  ┌──┬────┴─┐                          ┌──┬────┴─┐                  ┌──┬────┴─┐         ┌──────┬┴                              ┌──┬────┴─┐         ┌──────┬┴                             
                                    i  SimpleType                      i  SimpleType              i  b  SimpleType                     i  b  SimpleType             i  b  SimpleType    t  SimpleType                          i  b  SimpleType    t  SimpleType                         
                                          ┌┴                                 ┌┴                            ┌┴                                   ┌┴                           ┌┴              ┌─┴                                        ┌┴              ┌─┴                              
                                       integer                            integer                       integer                              integer                      integer          float                                     integer          float                              
